assets/js/admin/woocommerce_admin.js

```diff
@@ -73,6 +83,19 @@ jQuery( function ( $ ) {
 			}
 		})
 
+		.on( 'keyup', '.wc_input_strict_price[type=text]', function() {
+			var regex    = new RegExp( '[^0-9\\' + woocommerce_admin.mon_decimal_point + ']+', 'gi' );
```

Comment:
And if you are escaping `\'` who will escape the character returned by `woocommerce_admin.mon_decimal_point`?